draft/XXXX-multinetjs-autopublish/README.md

```diff
@@ -0,0 +1,102 @@
+linkages between MultinetJS and the client codebase.
+
+While this procedure works well in local development, it leaves CI out. In
+particular, since the Netlify build previews of the client rely on published
+versions of MultinetJS, build previews depending on a development version of
+MultinetJS simply cannot be built.
+
+A procedure for automated or semi-automated publishing of MultinetJS, even for
+development versions, would fix this issue. It would also eliminate the manual
+step currently needed to perform releases to NPM.
+
+## Proposals
+
+The proposals here use GitHub Actions to drive publishing pipelines to NPM on
+both pull request branches, and on master. Details for each follow.
+
+### Development Releases
+
+The idea here is to automatically publish development versions based on a PR
+branch to NPM automatically using CI. On each push from a PR branch, the GH
+Actions workflow can run the tests, perform a build, and if successful, publish
+a prerelease version to NPM. The version number can be taken from the current
+value in `package.json`, appended with the first six characters of the current
+commit's Git SHA--for example `multinet-0.47.3-a1c0ef` ([Clause
+9](https://semver.org/#spec-item-9) of the semver spec stipulates that
+prerelease versions can be indicated by appending alphanumeric strings in this
+manner). The publishing would be done using an NPM token stored in a GitHub
+secret, arranging the process so that any developer can effectively create
+releases simply by pushing to a branch (or master).
+
+Alternatively, instead of automatically pushing these releases, they could
+instead be done manually via an NPM script. However, this would have the
+drawback of requiring distribution of credentials to the different developers.
+GitHub Actions can also be used to trigger such actions manually; see below.
+
+### Production Releases
+
+Tools such as
+[semantic-release](https://github.com/semantic-release/semantic-release) can be
+used to perform automatically versioned releases of NPM projects. However, that
+approach requires a specific, machine-readable commit message format that can be
+used by the tooling to compute the correct release version.
+
+In place of that heavier-handed method, which introduces a specific burden on
+all developers, we can simply perform a release to NPM on every merge to master,
+based upon the version number listed in `package.json`. This essentially
+requires human intervention to both compute the correct semantic version for a
+given branch, and update it manually in the source. However, this also allows a
+bit of extra control, allowing for *not* updating the version number in order to
+avoid creating a release.
+
+### Manual GitHub Actions via Workflow Dispatch
+
+GitHub Actions includes a
+[workflow_dispatch](https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/)
+mechanism that allows for a pushbutton to execute a workflow. We can use this to
```

Comment:
What does security look like for this? Can we lock this down to just admins?

Reply:
I assume this is how it works, but it's something we'll have to keep in mind as we implement this.